File: testing_ex/c_data_structures_general/Pilha/pilha_seq.c
#include "pilha_seq.h"

/**Cria uma Pilha vazia*/
void cria(tPilha *pilha) {
    pilha->topo = -1;
}

/** Verifica se a Pilha está vazia */
int vazia(tPilha pilha) {
    if (pilha.topo == -1){
        return 1;
    }else{
        return 0;
    }
}

/** Verifica se a Pilha está cheia */
int cheia(tPilha pilha) {
    if (pilha.topo == MAX-1){
        return 1;
    }else{
        return 0;
    }
}

/** Obtém o tamanho da Pilha */
int tamanho(tPilha P) {
    return (P.topo + 1);
}

/** Consulta o elemento do topo da Pilha.
    Retorna 0 se a pilha estiver vazia, caso contrário retorna 1. */
int top(tPilha pilha, int *dado) {
    if (vazia(pilha)) {
            return 0; /* pilha vazia */
    }
    *dado = pilha.dados[pilha.topo];
    return 1;
}

/** Insere um elemento no topo da pilha.
    Retorna 0 se a pilha estiver cheia.
    Caso contrário retorna 1 */
int push(tPilha *pilha, int dado) {
    if (cheia(*pilha)){
            return 0;  /* pilha cheia */
    }
    else {
        pilha->topo++;
        pilha->dados[pilha->topo] = dado;
        return 1;
    }
}

/** Retira o elemento do topo da pilha.
    Retorna 0 se a pilha estiver vazia.
    Caso contrário retorna 1 */
int pop(tPilha *pilha, int *dado ) {
    if (vazia(*pilha)){
            return 0; /* Pilha vazia */
    }
    else {
            *dado = pilha->dados[pilha->topo];
            (pilha->topo)--;
            return 1;
    }
}

File: PS/PilhaSeq.py
class PilhaSeq:
    def __init__(self, TAM_MAX):
        self.TAM_MAX = TAM_MAX
        self.dados = []

    def vazia(self):
        return len(self.dados) == 0

    def cheia(self):
        return len(self.dados) == self.TAM_MAX

    def tamanho(self):
        return len(self.dados)

    def top(self):
        if self.vazia():
            return -1
        return self.dados[-1]

    def push(self, valor):
        if self.cheia():
            return False
        self.dados.append(valor)
        return True

    def pop(self):
        if self.vazia():
            return -1
        return self.dados.pop()